clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2518,13 +2517,12 @@ void handleFailure(Throwable throwable) {
             .collect(Collectors.toSet());
     }
 
-    private boolean handleFindCoordinatorError(FindCoordinatorResponse response, KafkaFutureImpl<?> future) {
-        Errors error = response.error();
-        if (error.exception() instanceof RetriableException) {
+    private boolean handleGroupRequestError(Errors error, KafkaFutureImpl<?> future) {
+        if (error == Errors.COORDINATOR_LOAD_IN_PROGRESS || error == Errors.COORDINATOR_NOT_AVAILABLE) {
```

Comment:
I'm guessing the reason you changed this is that NOT_COORDINATOR is considered a retriable exception?

Reply:
yeah. As you mentioned in the KAFKA-8341 JIRA, we may need  additional logic (find new coordinator?) to handle NOT_COORDINATOR error. For now, NOT_COORDINATOR treated as non-retriable and fail with exception.